runtime/src/accounts_db.rs

```diff
@@ -6245,13 +6245,29 @@ impl AccountsDb {
 
     /// add all items from 'skipped_rewrites' to 'hashes' where the pubkey doesn't already exist in 'hashes'
     fn extend_hashes_with_skipped_rewrites(
+        &self,
         hashes: &mut Vec<(Pubkey, Hash)>,
         skipped_rewrites: &Rewrites,
     ) {
         let mut skipped_rewrites = skipped_rewrites.read().unwrap().clone();
         hashes.iter().for_each(|(key, _)| {
             skipped_rewrites.remove(key);
         });
+
+        if self.filler_accounts_enabled() {
+            // simulate the time we would normally spend hashing the filler accounts
+            // this is an over approximation but at least takes a stab at simulating what the validator would spend time doing
+            let _ = AccountsHash::accumulate_account_hashes(
+                skipped_rewrites
+                    .iter()
+                    .map(|(k, v)| (*k, *v))
+                    .collect::<Vec<_>>(),
+            );
+
+            // filler accounts do not get their updated hash values hashed into the delta hash
+            skipped_rewrites.retain(|key, _| !self.is_filler_account(key));
+        }
```

Comment:
Is this needed for filler accounts to get a non-zero rent epoch? I'm reading it as orthogonal, so I've likely missed something.

Reply:
well, now that we are skipping rewrites on filler accounts, they are correctly in the list of skipped rewrites. We normally add skipped rewrite hashes into the bank hash. But, filler accounts need to NOT be added to the bank hash. So, the 'retain' removes filler accounts so they don't cause us to get the wrong bank hash.
The first code is an attempt to spend the time calculating the hash on the rewrites (which should be dominated by filler accounts) so that when we are using filler accounts (which are only a debug feature for perf testing), we more closely simulate how long we spend freezing a bank and calculating a bank hash. This first accumulate could be its own pr.

Reply:
I'll pull the first part out.

Reply:
Ah yes, the
```
            // filler accounts do not get their updated hash values hashed into the delta hash
            skipped_rewrites.retain(|key, _| !self.is_filler_account(key));
```
needs to stay.

Reply:
#25387